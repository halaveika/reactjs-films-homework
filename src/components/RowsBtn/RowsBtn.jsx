import React from 'react';
import './RowsBtn.scss';

export default function RowsBtn() {
  return (
 <div className="rows-btn">
   <div></div>
   <div></div>
   <div></div>
   <div></div>
 </div>
  );
}


