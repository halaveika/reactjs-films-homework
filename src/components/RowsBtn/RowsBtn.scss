.rows-btn {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  align-content: space-between;
  width: 20px;
  height: 20px;
  cursor: pointer;

  & div {
    width: 8px;
    height: 8px;
    background-color: #ced1d4;
  }

  &:hover div,
  &:active div {
    background-color: #1890ff;
  }
}
