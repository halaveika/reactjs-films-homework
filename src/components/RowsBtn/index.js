import RowsBtn from './RowsBtn';

export default RowsBtn;
