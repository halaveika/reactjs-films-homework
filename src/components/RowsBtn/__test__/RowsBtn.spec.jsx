import React from 'react';
import { create, act } from 'react-test-renderer';
import RowsBtn from '../RowsBtn';

const setListFlexDirection = jest.fn();

describe('test RowsBtn component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const component = create(
    <RowsBtn
      setListFlexDirection={setListFlexDirection}
      isRow={false}
    />,
  );

  it('should render RowsBtn component no active', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle onClick event', () => {
    const { root } = component;
    const clickable = root.findByProps({ className: 'rows-btn' });
    act(() => { clickable.props.onClick(); });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(setListFlexDirection).toBeCalled();
  });

  it('should render RowsBtn component active', () => {
    const component2 = create(
      <RowsBtn
        setListFlexDirection={setListFlexDirection}
        isRow
      />,
    );
    const tree = component2.toJSON();
    expect(tree).toMatchSnapshot();
    const { root } = component2;
    const button = root.findByProps({ className: 'rows-btn_active' });
    expect(button).toBeTruthy();
  });
});
