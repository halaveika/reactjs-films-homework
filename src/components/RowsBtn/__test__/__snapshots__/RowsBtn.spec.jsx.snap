// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RowsBtn component should handle onClick event 1`] = `
<div
  className="rows-btn"
  onClick={[Function]}
>
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`test RowsBtn component should render RowsBtn component 1`] = `
<div
  className="rows-btn"
  onClick={[Function]}
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
